python-sdk/tutorials/automl-with-azureml/ONNX_batch_model_generator_automl_for_images.py

```diff
@@ -0,0 +1,200 @@
+                              model_settings, model_type, device):
+    """
+    Get the onnx batch model and verify the schema.
+
+    :param batch_size: Batch size needed for batch inference
+    :type batch_size: int
+    :param height_onnx: Height of the image to be used for ONNX model
+    :type height_onnx: int
+    :param width_onnx: Width of the image to be used for ONNX model
+    :type width_onnx: int
+    :param task_type: Type of vision task from image classification, object detection, segmentation
+    :type task_type: str
+    :param best_child_run: Best child run object of an experiment
+    :type best_child_run: azureml.core.run.Run
+    :param model_settings: Model parameters with key, values
+    :type model_settings: dict
+    :param model_type: Model name
+    :type model_type: str
+    :param device: Device on which the model has to be generated
+    :type device: str
+    """
+
+    # download the pytorch model weights
+    torch_model_path = 'model.pt'
+    best_child_run.download_file(name='train_artifacts/model.pt', output_file_path=torch_model_path)
+    # load the model wrapper
+    model_wrapper = load_model(task_type, torch_model_path, **model_settings)
+    onnx_model_file_path = './outputs/model_' + str(batch_size) + '.onnx'
+    batch_model = False  # to check whether the user is not generating the default onnx model
+
+    if model_type == 'faster_rcnn':
+
+        if (batch_size > 1) or (height_onnx != 600) or (width_onnx != 800):
+            batch_model = True
+            input_names = ['input']
+            od_output_names = ['boxes', 'labels', 'scores']
+            output_names = [name + "_" + str(sample_id) for sample_id in range(batch_size) for name in od_output_names]
+            dynamic_axes = dict()
+            dynamic_axes['input'] = {0: 'batch'}
+            for output in output_names:
+                dynamic_axes[output] = {0: 'prediction'}
+
+            dummy_input = torch.randn(batch_size, 3, height_onnx, width_onnx).to(device)
+            model_wrapper.disable_model_transform()
+            model = model_wrapper.model
+
+    elif model_type == 'yolo':
+
+        if (batch_size > 1) or (height_onnx != 640) or (width_onnx != 640):
+            batch_model = True
+            input_names = ['input']
+            output_names = ['output']
+            dynamic_axes = dict()
+            dynamic_axes = {'input': {0: 'batch'}, 'output': {0: 'batch', 1: 'boxes'}}
+
+            dummy_input = torch.randn(batch_size, 3, height_onnx, width_onnx).to(device)
+            model = model_wrapper.model
+            for k, m in model.named_modules():
+                if isinstance(m, Conv) and isinstance(m.act, torch.nn.Hardswish):
+                    m.act = Hardswish()
+
+    elif model_type == 'mask_rcnn':
+
+        if (batch_size > 1) or (height_onnx != 600) or (width_onnx != 800):
+            batch_model = True
```

Comment:
can we refactor the code to reuse this for both od and segmentation?

Reply:
there are many differences between the two except for if condition